packages/bsky/src/api/app/bsky/notification/listActivitySubscriptions.ts

```diff
@@ -0,0 +1,123 @@
+import { mapDefined } from '@atproto/common'
+import { AppContext } from '../../../../context'
+import { HydrateCtx, Hydrator } from '../../../../hydration/hydrator'
+import { Server } from '../../../../lexicon'
+import { QueryParams } from '../../../../lexicon/types/app/bsky/notification/listActivitySubscriptions'
+import {
+  HydrationFnInput,
+  PresentationFnInput,
+  RulesFnInput,
+  SkeletonFnInput,
+  createPipeline,
+} from '../../../../pipeline'
+import { uriToDid as didFromUri } from '../../../../util/uris'
+import { Views } from '../../../../views'
+import { clearlyBadCursor, resHeaders } from '../../../util'
+
+export default function (server: Server, ctx: AppContext) {
+  const listActivitySubscriptions = createPipeline(
+    skeleton,
+    hydration,
+    noBlocks,
+    presentation,
+  )
+  server.app.bsky.notification.listActivitySubscriptions({
+    auth: ctx.authVerifier.standard,
+    handler: async ({ params, auth, req }) => {
+      const viewer = auth.credentials.iss
+      const labelers = ctx.reqLabelers(req)
+      const hydrateCtx = await ctx.hydrator.createContext({
+        labelers,
+        viewer,
+      })
+
+      const result = await listActivitySubscriptions(
+        { ...params, hydrateCtx: hydrateCtx.copy({ viewer }) },
+        ctx,
+      )
+
+      return {
+        encoding: 'application/json',
+        body: result,
+        headers: resHeaders({ labelers: hydrateCtx.labelers }),
+      }
+    },
+  })
+}
+
+const skeleton = async (input: SkeletonFnInput<Context, Params>) => {
+  const { params, ctx } = input
+  const actorDid = params.hydrateCtx.viewer
+  if (clearlyBadCursor(params.cursor)) {
+    return { actorDid, dids: [] }
+  }
+  const { dids, cursor } =
+    await ctx.hydrator.dataplane.getActivitySubscriptions({
+      actorDid: params.hydrateCtx.viewer,
+      limit: params.limit,
+      cursor: params.cursor,
+    })
+  return {
+    actorDid,
+    dids,
+    cursor: cursor || undefined,
+  }
+}
+
+const hydration = async (
+  input: HydrationFnInput<Context, Params, SkeletonState>,
+) => {
+  const { ctx, params, skeleton } = input
+  const { dids } = skeleton
+  const followState = await ctx.hydrator.hydrateProfiles(
+    dids,
+    params.hydrateCtx,
+  )
+  return followState
+}
+
+const noBlocks = (input: RulesFnInput<Context, Params, SkeletonState>) => {
+  const { skeleton, params, hydration, ctx } = input
+  const viewer = params.hydrateCtx.viewer
+  skeleton.dids = skeleton.dids.filter((followUri) => {
+    const followerDid = didFromUri(followUri)
+    return (
+      !hydration.followBlocks?.get(followUri) &&
```

Comment:
I believe `followBlocks` are used for the case when you're just looking at follow records of other users, as in `getFollows` or `getFollowers` endpoints.